cloud/kubernetes/helm/templates/_helpers.tpl

```diff
@@ -0,0 +1,226 @@
+        {{- else }}
+      spec:
+        accessModes: {{ .Values.volumes.accessModes }}
+        volumeName: {{ .Values.volumes.snVolume1 }}
+        {{- end }}
+        resources:
+          requests:
+            storage: {{ .Values.volumes.storage }}
+{{- end }}
+{{- end -}}
+
+{{- define "sharedNothing.inferencesecond" -}}
+{{- if eq .Values.isType "sharedNothing" }}
+  volumeClaimTemplates:
+    - metadata:
+        name: {{ .Values.volumes.name }}
+        {{- if ne .Values.cloudProvider "openshift" }}
+        annotations:
+          volume.beta.kubernetes.io/storage-class: {{ .Values.volumes.storageClass }}
+      spec:
+        accessModes: {{ .Values.volumes.accessModes }}
+        {{- else }}
+      spec:
+        accessModes: {{ .Values.volumes.accessModes }}
+        volumeName: {{ .Values.volumes.snVolume2 }}
+        {{- end }}
+        resources:
+          requests:
+            storage: {{ .Values.volumes.storage }}
+{{- end }}
+{{- end -}}
+
+{{- define "sharedAll.first" -}}
+{{- if eq .Values.isType "sharedAll" }}
+- name: {{ .Values.database.envname1 }}
+  valueFrom:
+    configMapKeyRef:
+      name: {{ .Values.database.name }}
+      key: {{ .Values.database.value1 }}
+- name: {{ .Values.database.envname2 }}
+  valueFrom:
+    configMapKeyRef:
+      name: {{ .Values.database.name }}
+      key: {{ .Values.database.value2 }}
+- name: {{ .Values.database.envname3 }}
+  valueFrom:
+    configMapKeyRef:
+      name: {{ .Values.database.name }}
+      key: {{ .Values.database.value3 }}
+- name: {{ .Values.database.envname4 }}
+  valueFrom:
+    configMapKeyRef:
+      name: {{ .Values.database.name }}
+      key: {{ .Values.database.value4 }}
+- name: {{ .Values.database.envname5 }}
+  valueFrom:
+    configMapKeyRef:
+      name: {{ .Values.database.name }}
+      key: {{ .Values.database.value5 }}
+- name: {{ .Values.database.envname6 }}
+  valueFrom:
+    configMapKeyRef:
+      name: {{ .Values.database.name }}
+      key: {{ .Values.database.value6 }}
+{{- end }}
+{{- end -}}
+
+{{- define "sharedAll.second" -}}
+{{- if eq .Values.isType "sharedAll" }}
+volumeMounts:
+  - mountPath: {{ .Values.volumes.samountPath }}
+    name: {{ .Values.volumes.saname }}
+{{- end }}
+{{- end -}}
+
+{{- define "sharedAll.third" -}}
+{{- if eq .Values.isType "sharedAll" }}
+  volumeClaimTemplates:
+    - metadata:
+        name: {{ .Values.volumes.saname }}
+        {{- if ne .Values.cloudProvider "openshift" }}
+        annotations:
+          volume.beta.kubernetes.io/storage-class: {{ .Values.volumes.sastorageClass }}
+      spec:
+        accessModes: {{ .Values.volumes.accessModes }}
+        {{- else }}
+      spec:
+        accessModes: {{ .Values.volumes.accessModes }}
+        volumeName: {{ .Values.volumes.saVolume }}
+        {{- end }}
+        resources:
+          requests:
+            storage: {{ .Values.volumes.sastorage }}
+{{- end }}
+{{- end -}}
+
+{{- define "commonconfigmap.data" -}}
```

Comment:
Add single line comments for each of the groups indicating what are these properties for.